### PR TITLE
Add reflection to grpc api

### PR DESCRIPTION
**Pull request by venezia**
This will add reflection as an option for the grpc api.
It also makes a small correction in the documentation - dex will not create self signed certificates if certs are not provided, rather it runs in plaintext mode.

Does not fix, but this work was discussed in #1499

### CODE REVIEW

**Comment by venezia** [comment]
travis wasn't able to [connect to github during the test](https://travis-ci.org/dexidp/dex/builds/568556609?utm_source=github_status&utm_medium=notification) - unsure how to get travis to try again

File: cmd/dex/serve.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 	"crypto/x509"
 	"errors"
 	"fmt"
+	"google.golang.org/grpc/reflection"
```
Comment (srenatus) [inline cmd/dex/serve.go]: Please put that into a section with the other 3rd-party imports :)

Comment (venezia) [reply cmd/dex/serve.go]: Sorry, I IDE did that automatically for me and I wasn't paying attention - sure

**Comment by venezia** [comment]
> LGTM, thanks for this! 😃 Would you mind squashing this into two commits -- one for the code, one for the revendor? That would also trigger travis again, I hope 🤞

Sure, but just know I was only following [dex's guidelines](https://github.com/dexidp/dex/blob/master/Documentation/dev-dependencies.md#composing-commits)

That being said do you want, in the future, for commits involving dependencies to be in the same commit as regular code changes?

**Comment by srenatus** [comment]
@venezia thanks for pointing out those guidelines. I had forgotten about that. They're quite agreeable -- at any rate, could you amend your commits to satisfy them? 😉 
Like
1. `cmd/dex: make reflection configurable`
2. `vendor: revendor`

and the third commit can be squashed into the first one?

**Comment by srenatus** [comment]
If you want to split commits more (while adhering to the guidelines), please do so -- I'm sorry for misleading you here 😉

**Comment by venezia** [comment]
> If you want to split commits more (while adhering to the guidelines), please do so -- I'm sorry for misleading you here 😉

no, this is what happens when I misread a comment while still waking up.  I think here one commit is just fine, sometimes if it is bigger maybe more than one commit is better (if they're doing separate things, etc.) but that's for another day. 

Regardless, things should be squashed and commented better now.

**Pull request merged**